torch/csrc/distributed/c10d/FlightRecorder.hpp

```diff
@@ -0,0 +1,257 @@
+#pragma once
+
+// TODO: Make Fligth Recorder device agnostic
+#ifdef USE_C10D_NCCL
+
+#include <cstdio>
+#include <cstdlib>
+
+#include <memory>
+#include <mutex>
+
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAEvent.h>
+#include <c10/util/Exception.h>
+#include <torch/csrc/distributed/c10d/TraceUtils.h>
+#include <optional>
+
+
+namespace c10d {
+
+#define DEFINE_CONSTANT(name, value) \
+  static c10::IValue name = value;   \
+  static std::string name##_str = value;
+// Update whenever changing contents or formatting of the dump
+// (minor when adding fields, major when changing existing fields)
+// Also update both JSON and Pickle dumps to make use of the newly defined
+// field(s).
+DEFINE_CONSTANT(version_val, "2.4")
```

Comment:
Can you bump up the version to maybe "3.0", so that we know the dump before and after this change since this is a big change.

Reply:
Hmm, there is no substantial value change here.